drv/cosmo-seq-server/src/main.rs

```diff
@@ -664,6 +676,35 @@ impl NotificationHandler for ServerImpl {
             _ => (),
         }
 
+        // If we're powered up, detect powering down
+        if matches!(self.state, PowerState::A0 | PowerState::A0PlusHP) {
```

Comment:
Something that I observed empirically is that the transition into A2 causes the reset/powerok edge counters as maintained by the FPGA to increment; going back to A0 causes the register to reset itself.

I believe that this `if` statement should protect against a vicious reset loop when going to A2, but it may be worth a comment mentioning it, and I wonder if we should move the count/IFR field reset logic from here and to somewhere else (e.g., the transition from A2 to A0? Dunno off the top of my head).

Reply:
Clearing the flags in the A2 -> A0 transition makes sense to me, done in 9aa7d90bffb9